src/shapes/Group.ts

```diff
@@ -136,9 +139,7 @@ export class Group
     });
 
     // perform initial layout
-    const layoutManager =
-      // not destructured on purpose here.
-      options.layoutManager || new LayoutManager();
+    this.layoutManager = layoutManager;
```

Comment:
i removed destructuring because is extremely noisy in the constructor, please don't change this since the change was fresh and commented. the rest operator on the function parameter becomes larger code and also slower. We do not get any benefit from extracting the layoutManager and assign it as a default in the constructor, since the layout manager is not used in the super class of the group.

Reply:
I don't mind about destruction, though having super setting values has become something I hate. I have encountered a number of hard to debug issues caused by fabric's usage of it. But that is a different topic.
I don't care about destructing or not but your change not to destruct the object introduced a regression and confusion.
It has to be set on the instance before calling performLayout or else it will not subscribe the objects.
Now, after you made group bind a layout manager I think we should remove that check all together but we need to consider how it will affect sharing a layout manager between groups and if we allow it or not.
So yes I prefer destructing because there is less ambiguity. It is just that a naive syntax introduced bugs.
You did not write why it was not destructed on purpose.

Reply:
To conclude, the syntax is 99% of the times something I can live with and bugs suck

Reply:
> Is there any case in which we end up executing that line with a layout manager and a group that are not bound together?

That is what I mentioned

> Now, after you made group bind a layout manager I think we should remove that check all together but we need to consider how it will affect sharing a layout manager between groups and if we allow it or not.